### PR TITLE
fix!: Removal of deprecated methods

### PR DESCRIPTION
**Pull request by antusus**

### CODE REVIEW

File: doc/upgrades/3.x.x to 4.x.x.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+3. `com.box.sdk.BoxEvent.getType` - replaced with `com.box.sdk.BoxEvent.getEventType`
+4. `com.box.sdk.BoxEvent.Type` - replaced with `com.box.sdk.BoxEvent.EventType`
+5. `com.box.sdk.BoxItem.createSharedLink` - removed methods that do not use `com.box.sdk.sharedlink.BoxSharedLinkRequest`, please ones that use it
+6. `com.box.sdk.BoxFile.createSharedLink`- removed methods that do not use `com.box.sdk.sharedlink.BoxSharedLinkRequest`, please ones that use it
+7. `com.box.sdk.BoxFolder.createSharedLink`- removed methods that do not use `com.box.sdk.sharedlink.BoxSharedLinkRequest`, please ones that use it
+8. `com.box.sdk.BoxWebLink.createSharedLink` - removed methods that do not use `com.box.sdk.sharedlink.BoxSharedLinkRequest`, please ones that use it
+9. `com.box.sdk.BoxFile.uploadVersion` - methods are replaced with `com.box.sdk.BoxFile.uploadNewVersion`
+10. `com.box.sdk.BoxFolder.search` - is replaced with class `com.box.sdk.BoxSearch`
+11. `com.box.sdk.BoxGroup` - removed mthods that use `com.box.sdk.BoxGroupMembership.Role`, please use ones that uses `com.box.sdk.BoxGroupMembership.GroupRole`
```
Comment (mwwoda) [inline doc/upgrades/3.x.x to 4.x.x.md:31]: Typo
```suggestion
11. `com.box.sdk.BoxGroup` - removed methods that use `com.box.sdk.BoxGroupMembership.Role`, please use ones that uses `com.box.sdk.BoxGroupMembership.GroupRole`
```

File: doc/upgrades/3.x.x to 4.x.x.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+# Removed deprecated methods and classes
+1. `com.box.sdk.BoxAPIRequest.BoxAPIRequest(java.net.URL, java.lang.String)`
+2. `com.box.sdk.BoxUser.moveFolderToUser` - this method is removed as this operation is not allowed by API. You can only transfer root folder to another user using `com.box.sdk.BoxUser.transferContent`
+
+# Replaced methods and classes
+1. `com.box.sdk.BoxDeveloperEditionAPIConnection` - removed all constructors that do not accept `com.box.sdk.IAccessTokenCache`. Use versions that accepts cache parameter
+2. `com.box.sdk.BoxDeveloperEditionAPIConnection#getAppUserConnection` - replaced with com.box.sdk.BoxDeveloperEditionAPIConnection#getUserConnection
+3. `com.box.sdk.BoxEvent.getType` - replaced with `com.box.sdk.BoxEvent.getEventType`
+4. `com.box.sdk.BoxEvent.Type` - replaced with `com.box.sdk.BoxEvent.EventType`
+5. `com.box.sdk.BoxItem.createSharedLink` - removed methods that do not use `com.box.sdk.sharedlink.BoxSharedLinkRequest`, please ones that use it
```
Comment (mwwoda) [inline doc/upgrades/3.x.x to 4.x.x.md:25]: `removed methods that do not use...please ones that use it`
I guess this sentence is missing some words. Please also check this in other places

**Review by mwwoda — COMMENTED** [review]
Just some spelling issues

**Review by mwwoda — APPROVED** [review]

File: doc/upgrades/3.x.x to 4.x.x.md [hunk 3]
```diff
@@ -0,0 +1,174 @@
+# Migration guide for versions 3.x.x -> 4.x.x
+
+The most important change this release includes is the replacement of the HTTP library from a native one to
+OkHttp which allows SDK to
```
Comment (bszwarc) [inline doc/upgrades/3.x.x to 4.x.x.md:4]:
```suggestion
OkHttp which allows SDK to:
```

File: doc/upgrades/3.x.x to 4.x.x.md [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+
+# Configuration changes
+
+## MaxRequestAttempts
+The `MaxRequestAttempts` are removed from `com.box.sdk.BoxGlobalSettings` and replaced with `MaxRetryAttempts`. 
+However, if you have API connection stored with the deprecated `MaxRequestAttempts` value it will be restored into `MaxRetryAttempts`.
+
+# Removed deprecated methods and classes
+1. `com.box.sdk.BoxAPIRequest.BoxAPIRequest(java.net.URL, java.lang.String)` - use constuctor that accepts `com.box.sdk.BoxAPIConnection`.
+2. `com.box.sdk.BoxUser.moveFolderToUser` - this method is removed as this operation is not allowed by API. You can only transfer root folder to another user using `com.box.sdk.BoxUser.transferContent`
```
Comment (bszwarc) [inline doc/upgrades/3.x.x to 4.x.x.md:18]:
```suggestion
2. `com.box.sdk.BoxUser.moveFolderToUser` - this method is removed as this operation is not allowed by API. You can only transfer root folder to another user using `com.box.sdk.BoxUser.transferContent`.
```

File: doc/upgrades/3.x.x to 4.x.x.md [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+## Shared links
+Removed variant of methods that do not use `com.box.sdk.sharedlink.BoxSharedLinkRequest`:
+1. `com.box.sdk.BoxItem.createSharedLink`
+2. `com.box.sdk.BoxFile.createSharedLink`
+3. `com.box.sdk.BoxFolder.createSharedLink`
+4. `com.box.sdk.BoxWebLink.createSharedLink`
+
+### Example
+
+If you want to create a shared link first create `com.box.sdk.sharedlink.BoxSharedLinkRequest` example:
```
Comment (bszwarc) [inline doc/upgrades/3.x.x to 4.x.x.md:31]:
```suggestion
If you want to create a shared link, first create `com.box.sdk.sharedlink.BoxSharedLinkRequest`:
```

File: doc/upgrades/3.x.x to 4.x.x.md [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
+```
+
+### Documentation
+More details can be found:
+1. [com.box.sdk.BoxFile.createSharedLink](../files.md#create-a-shared-link)
+2. [com.box.sdk.BoxFolder.createSharedLink](../folders.md#create-a-shared-link)
+3. [com.box.sdk.BoxWebLink.createSharedLink](../weblinks.md#create-shared-link)
+
+## Retention Policies
+Removed variant of methods that were not using `com.box.sdk.BoxRetentionPolicy.BoxRetentionPolicyAction`
```
Comment (bszwarc) [inline doc/upgrades/3.x.x to 4.x.x.md:50]:
```suggestion
Removed variant of methods that were not using `com.box.sdk.BoxRetentionPolicy.BoxRetentionPolicyAction`:
```

File: doc/upgrades/3.x.x to 4.x.x.md [hunk 7]
```diff
@@ -0,0 +1,174 @@
...
+If you want to create a finitie retention policy:
+```java
+BoxRetentionPolicy.createFinitePolicy(api, "My 30 days retention policy", 30, BoxRetentionPolicyAction.PermanentlyDelete);
+```
+
+### Documentation
+You can read more on creating retention policies [here](../retention_policies.md#create-retention-policy).
+
+## Enterprise Events
+Removed variant of methods that were not using `com.box.sdk.EnterpriseEventsRequest`
```
Comment (bszwarc) [inline doc/upgrades/3.x.x to 4.x.x.md:63]:
```suggestion
Removed variant of methods that were not using `com.box.sdk.EnterpriseEventsRequest`:
```

File: doc/upgrades/3.x.x to 4.x.x.md [hunk 8]
```diff
@@ -0,0 +1,174 @@
...
+1. `com.box.sdk.EventLog.getEnterpriseEvents`
+### Example
+If you want to get historical enterprise events first create `com.box.sdk.EnterpriseEventsRequest`:
```
Comment (bszwarc) [inline doc/upgrades/3.x.x to 4.x.x.md:66]:
```suggestion
If you want to get historical enterprise events, first create `com.box.sdk.EnterpriseEventsRequest`:
```

File: doc/upgrades/3.x.x to 4.x.x.md [hunk 9]
```diff
@@ -0,0 +1,174 @@
...
+
+BoxSearch boxSearch = new BoxSearch(api);
+boxSearch.searchRange(offsetValue, limitValue, searchParams);
+```              
+
+### Documentation
+You can read more on search [here](../search.md).
+
+## BoxGroup
+All methods that use `com.box.sdk.BoxGroupMembership.Role` were removed, please use ones that uses `com.box.sdk.BoxGroupMembership.GroupRole`.
```
Comment (bszwarc) [inline doc/upgrades/3.x.x to 4.x.x.md:118]:
```suggestion
All methods that use `com.box.sdk.BoxGroupMembership.Role` were removed. Use ones that use `com.box.sdk.BoxGroupMembership.GroupRole`.
```
We generally don't use `please` in tech documentation :)

File: doc/upgrades/3.x.x to 4.x.x.md [hunk 10]
```diff
@@ -0,0 +1,174 @@
...
+    .unsharedDate(unsharedDate);
+
+BoxSharedLink sharedLink = file.createSharedLink(sharedLinkRequest);
+```
+
+### Documentation
+More details can be found:
+1. [com.box.sdk.BoxFile.createSharedLink](../files.md#create-a-shared-link)
+2. [com.box.sdk.BoxFolder.createSharedLink](../folders.md#create-a-shared-link)
+3. [com.box.sdk.BoxWebLink.createSharedLink](../weblinks.md#create-shared-link)
```
Comment (bszwarc) [inline doc/upgrades/3.x.x to 4.x.x.md:47]:
```suggestion
1. [Create shared link for file](../files.md#create-a-shared-link)
2. [Create shared link for folder](../folders.md#create-a-shared-link)
3. [Create shared link for web link](../weblinks.md#create-shared-link)
```

File: doc/upgrades/3.x.x to 4.x.x.md [hunk 11]
```diff
@@ -0,0 +1,174 @@
...
+BoxSharedLinkRequest sharedLinkRequest = new BoxSharedLinkRequest()
+    .access(OPEN)
+    .permissions(true, true)
+    .unsharedDate(unsharedDate);
+
+BoxSharedLink sharedLink = file.createSharedLink(sharedLinkRequest);
+```
+
+### Documentation
+More details can be found:
```
Comment (bszwarc) [inline doc/upgrades/3.x.x to 4.x.x.md:44]:
```suggestion
For details on shared links, see:
```

File: doc/upgrades/3.x.x to 4.x.x.md [hunk 12]
```diff
@@ -0,0 +1,174 @@
...
+BoxDeveloperEditionAPIConnection api = new BoxDeveloperEditionAPIConnection(
+       USER_ID,
+       DeveloperEditionEntityType.USER,
+       boxConfig,
+       accessTokenCache
+);
+```
+
+### Documentation
+Check [README.md](../../README.md#boxdevelopereditionapiconnectionasenterpriseuser).
```
Comment (bszwarc) [inline doc/upgrades/3.x.x to 4.x.x.md:96]:
```suggestion
You can read more on BoxDeveloperEditionAPIConnection [here](../../README.md#boxdevelopereditionapiconnectionasenterpriseuser).
```

File: doc/upgrades/3.x.x to 4.x.x.md [hunk 13]
```diff
@@ -0,0 +1,174 @@
...
+BoxGroupMembership.Info membershipInfo = membership.getInfo();
+membershipInfo.setGroupRole(BoxGroupMembership.GroupRole.MEMBER);
+membership.updateInfo(membershipInfo);
+```
+
+### Documentation
+You can read more on groups [here](../groups.md).
+
+## MetadataTemplate
+In `com.box.sdk.MetadataTemplate` methods that do not use `com.box.sdk.MetadataQuery` where removed.
```
Comment (bszwarc) [inline doc/upgrades/3.x.x to 4.x.x.md:146]:
```suggestion
In `com.box.sdk.MetadataTemplate` methods that do not use `com.box.sdk.MetadataQuery` were removed. 
```

File: doc/upgrades/3.x.x to 4.x.x.md [hunk 14]
```diff
@@ -0,0 +1,174 @@
...
+1. [com.box.sdk.BoxFile.createSharedLink](../files.md#create-a-shared-link)
+2. [com.box.sdk.BoxFolder.createSharedLink](../folders.md#create-a-shared-link)
+3. [com.box.sdk.BoxWebLink.createSharedLink](../weblinks.md#create-shared-link)
+
+## Retention Policies
+Removed variant of methods that were not using `com.box.sdk.BoxRetentionPolicy.BoxRetentionPolicyAction`
+1. `com.box.sdk.BoxRetentionPolicy.createFinitePolicy`
+
+### Example
+If you want to create a finitie retention policy:
```
Comment (bszwarc) [inline doc/upgrades/3.x.x to 4.x.x.md:54]:
```suggestion
If you want to create a finite retention policy:
```

File: doc/upgrades/3.x.x to 4.x.x.md [hunk 15]
```diff
@@ -0,0 +1,174 @@
...
+
+### Documentation
+You can read more on search [here](../search.md).
+
+## BoxGroup
+All methods that use `com.box.sdk.BoxGroupMembership.Role` were removed, please use ones that uses `com.box.sdk.BoxGroupMembership.GroupRole`.
+Also `com.box.sdk.BoxGroupMembership.Role` was replaced with `com.box.sdk.BoxGroupMembership.GroupRole`.
+
+### Example
+To add nen member to a group with specific role:
```
Comment (bszwarc) [inline doc/upgrades/3.x.x to 4.x.x.md:122]:
```suggestion
To add a new member to a group with specific role:
```

**Review by bszwarc — COMMENTED** [review]
Several minor comments :)

**Review by bszwarc — APPROVED** [review]

**Pull request merged**